app/controllers/manage_metadata_controller.rb

```diff
@@ -24,25 +24,37 @@ def set_variable_information
     # if the variable is not found, try again because CMR might be a little slow to index if it is a newly published record
     attempts = 0
     while attempts < 20
-      variables_search_response = cmr_client.get_variables(concept_id: @concept_id)
+      variables_search_response = cmr_client.get_variables(concept_id: @concept_id, all_revisions: true)
 
       variable_data = if variables_search_response.success?
-                        variables_search_response.body['items'].first
+                        variables_search_response.body.fetch('items', [])
                       else
-                        {}
+                        [{}]
                       end
+      variable_data.sort! { |a, b| b['meta']['revision-id'] <=> a['meta']['revision-id'] }
 
-      break if !variable_data.nil? && variable_data.fetch('meta', {})['concept-id'] == @concept_id
+      # break if revisions is wrong
+      latest = variable_data.first
+
+      if !@revision_id.nil? && latest && latest['meta']['revision-id'].to_s != @revision_id.to_s
+        @old_revision = true
+      end
+
+      break if latest && !@revision_id
+      break if latest && latest['meta']['revision-id'] >= @revision_id.to_i && latest.fetch('meta', {})['concept-id'] == @concept_id
       attempts += 1
       sleep 0.05
     end
 
-    if variable_data.blank?
+    @revisions = variable_data
+    first_variable = variable_data.first
```

Comment:
is there a difference between `latest` and `first_variable`? if not, would it be better to just use one of them?

Reply:
no, they are the same, I'll change it